tony-proxy/src/main/java/com/linkedin/tonyproxy/ProxyServer.java

```diff
@@ -52,57 +52,40 @@ public void start() throws IOException {
 
     @Override
     public void run() {
-      try {
+      try (Socket server = new Socket(serverHost, serverPort);
+          final InputStream inFromClient = clientSocket.getInputStream();
+          final OutputStream outToClient = clientSocket.getOutputStream();
+          final InputStream inFromServer = server.getInputStream();
+          final OutputStream outToServer = server.getOutputStream();
```

Comment:
Code looks much cleaner with the try-with-resources construct!

Reply:
Yep